registry/storage/driver/gdrive/gdrive.go

```diff
@@ -0,0 +1,497 @@
+// object.
+func (d *driver) Move(ctx context.Context, sourcePath string, destPath string) error {
+	dstFile := &drive.File{}
+	file := d.getFile(sourcePath)
+	if file != nil {
+		dstFile.Name = destPath
+		_, err := d.driveService.Files.Update(file.Id, dstFile).Do()
+		if err != nil {
+			return err
+		}
+	} else {
+		return storagedriver.PathNotFoundError{Path: sourcePath}
+	}
+	return nil
+}
+
+// Delete recursively deletes all objects stored at "path" and its subpaths.
+func (d *driver) Delete(ctx context.Context, subPath string) error {
+	r, _ := d.driveService.Files.List().
+		Q(`"` + d.rootID + `" in parents`).Do()
+	pathFound := false
+	if r != nil && len(r.Files) > 0 {
+		for _, i := range r.Files {
+			if subPath != "/" && strings.HasPrefix(i.Name, subPath) {
+				pathFound = true
+				err := d.driveService.Files.Delete(i.Id).Do()
+				if err != nil {
+					log.Fatalf("Cannot delete the file:%s %s", i.Name, err)
+				}
+			}
+		}
+	}
+	if !pathFound {
+		return storagedriver.PathNotFoundError{Path: subPath}
+	}
+	return nil
+}
+
+// URLFor returns a URL which may be used to retrieve the content stored at the given path.
+// May return an UnsupportedMethodErr in certain StorageDriver implementations.
+func (d *driver) URLFor(ctx context.Context, path string, options map[string]interface{}) (string, error) {
+	return "", storagedriver.ErrUnsupportedMethod{}
+}
+
+type fileInfo struct {
+	driveFileInfo driveFileInfo
+	path          string
+}
+
+type driveFileInfo struct {
+	modifiedTime string
+	size         int64
+	isDir        bool
+}
+
+// Path provides the full path of the target of this file info.
+func (fi fileInfo) Path() string {
+	return fi.path
+}
+
+// Size returns current length in bytes of the file. The return value can
+// be used to write to the end of the file at path. The value is
+// meaningless if IsDir returns true.
+func (fi fileInfo) Size() int64 {
+	return fi.driveFileInfo.size
+}
+
+// ModTime returns the modification time for the file. For backends that
+// don't have a modification time, the creation time should be returned.
+func (fi fileInfo) ModTime() time.Time {
+	t, _ := time.Parse(time.RFC3339, fi.driveFileInfo.modifiedTime)
+	return t
+}
+
+// IsDir returns true if the path is a directory.
+func (fi fileInfo) IsDir() bool {
+	return fi.driveFileInfo.isDir
+}
+
+type fileWriter struct {
+	bytes     []byte
+	path      string
+	size      int
+	driver    *driver
+	closed    bool
+	committed bool
+	cancelled bool
+}
+
+func newFileWriter(bytes []byte, path string, size int, d *driver) *fileWriter {
+	return &fileWriter{
+		bytes:  bytes,
+		path:   path,
+		size:   size,
+		driver: d,
+	}
+}
+
+func (fw *fileWriter) Write(p []byte) (int, error) {
+	if fw.closed {
+		return 0, fmt.Errorf("already closed")
+	} else if fw.committed {
+		return 0, fmt.Errorf("already committed")
+	} else if fw.cancelled {
+		return 0, fmt.Errorf("already cancelled")
+	}
+
+	fw.bytes = append(fw.bytes, p...)
```

Comment:
This is just buffering everything into memory. This should write the data through to the backend.